draft-irtf-nwcrg-coding-and-congestion-01.txt

```diff
@@ -1,504 +1,528 @@
+
+   o  The server must be able to detect the "repaired packet" signal.
+      The base solution does not describe how the congestion control
+      reacts to such signal.
+
+   The rationale behind this solution is the following: the server has
+   more information on the congestion status and the application
+   characteristics.  Moreover, congestion control should not be split
+   between multiple entities, as otherwise non-optimized decisions may
+   be taken.
+
+   The proposed solution applies for coding in the transport and
+   application layers.  The proposed approach is in-line with the one in
+   [I-D.swett-nwcrg-coding-for-quic].  The proposed solution does not
+   apply for the interaction between coding under the transport layer
+   (i.e.  not end-to-end), such as coding for tunnels.
+
+5.  Server-side coding solutions
+
+   This section presents solutions for a server to add application
+   coding.
+
+5.1.  Coded packets without considering CWND progression
+
+   In this solution, the coded packets are sent on top of what is
+   allowed by a congestion window.  Examples of the solution could be
+   adding a given percentage of the congestion window as supplementary
+   packets or sending a given amount of coded packets at a given rate.
+   The redundancy flow can be decorrelated from the congestion control
+   that manages source packets: a secondary congestion control can be
+   introduced, such as in coupled congestion control for RTP media [I-D
+   .ietf-rmcat-coupled-cc].  An example would be to exploit a lower than
+   best-effort congestion control [RFC6297].
+
+   The advantage of such a solution is that coding would help in
+   challenging cases where transmission losses are persistent.
+
+   The drawback of such a solution is that it may result in coding
+   solutions being unfair towards non-coding solutions.  This solutions
+   may result in adding congestion in congested networks.
+
+5.2.  Coded packets driven by CWND progression
+
+   In this solution, the coded packets are sent within what the
+   congestion window allows, such as in [CTCP].  Examples of the
+   solution would be sending coded packets when there is no more data to
+   transmit or preferably send coded packets instead of the following
+   packets in the send buffer.
+
+
+Kuhn, Lochin, Michel & WeExpires July 11, 2020                  [Page 5]
+
+Internet-Draft           Coding and congestion              January 2020
+
+
+   The advantage of this solution is that it does not add more traffic
+   than the congestion window allows.  Indeed, all traffic (source and
+   redundancy) is controlled by one congestion control only and TCP
+   metrics for fairness can be indifferently applied in this case.
+
+   The main drawback is the decrease of goodput if coded packets are
+   sent but are not used at the client side.
+
+6.  Server-side reaction to repaired packet signals
+
+   [MICHAEL: I would remove this section entirely: it only adds to

```

Comment:
I am personnaly OK to remove this.

In my opinion, we can remove this sections and simply state that coding should not be used to artificially increase throughput by hiding loss signals the the CC on a link where losses could actually indicate the presence of congestion.

But then what about the solutions that protect TCP connections running on a link (e.g. satellite link) where losses should not be due to congestion ? These are solutions that improve drastically performances, are easy to deploy and are not harmful in a congestion point of view. What is your opinion about this ? I am not sure myself.

Reply:
Interesting to discuss those things!  These are cases where the environment is known so well that it is clear that hiding these losses won't create a problem. So we can say: don't generally hide losses UNLESS you know for sure that they are not due to congestion.  This also fits well in an implementation discussion structured around the 3 layering possibilities (above, inside, below).

Reply:
I think there are things in the current text that are valid points for the 3 layering possibilities.